src/api/fetchData.js

```diff
@@ -2,24 +2,29 @@ import { useState, useEffect } from 'react';
 
 export default function fetchData(url) {
     const [data, setData] = useState([]);
-    const [loading, setLoading] = useState(null);
+    const [loading, setLoading] = useState(true);
     const [error, setError] = useState('');
 
+    const rawgioApiKey = '';
+    const requestOptions = {
+        headers: {
+            'Target-URL': 'https://rawg.io',
+            Authorization: '', // cors-proxy expects this attribute, otherwise it fails
+        },
+    };
+
     useEffect(() => {
-        setLoading(true);
-        try {
-            fetch(url)
-                .then((res) => res.json())
-                .then((data) => setData(data));
-        } catch (e) {
-            throw new Error(
-                setError(
-                    `Cannot fetch the data, error with status ${response.status}`
-                )
-            );
-        } finally {
-            setLoading(false);
-        }
+        fetch(`${url}?key=${rawgioApiKey}`, requestOptions)
+            .then((res) => res.json())
+            .then((data) => setData(data))
+            .catch((error) => {
+                throw new Error(
```

Comment:
Maybe its better: 

.catch((err) => {
setError(
                        `Cannot fetch the data, error with status ${err}`
                    )
                throw new Error(error)
                
So you set the error in the state first and then return throw that error.